src/main/java/org/mskcc/cmo/publisher/pipeline/BatchConfiguration.java

```diff
@@ -0,0 +1,174 @@
+package org.mskcc.cmo.publisher.pipeline;
+
+import java.util.Map;
+import java.util.concurrent.Future;
+import org.mskcc.cmo.messaging.Gateway;
+import org.mskcc.cmo.publisher.pipeline.limsrest.LimsRequestListener;
+import org.mskcc.cmo.publisher.pipeline.limsrest.LimsRequestProcessor;
+import org.mskcc.cmo.publisher.pipeline.limsrest.LimsRequestReader;
+import org.mskcc.cmo.publisher.pipeline.limsrest.LimsRequestWriter;
+import org.springframework.batch.core.Job;
+import org.springframework.batch.core.Step;
+import org.springframework.batch.core.StepExecutionListener;
+import org.springframework.batch.core.configuration.annotation.EnableBatchProcessing;
+import org.springframework.batch.core.configuration.annotation.JobBuilderFactory;
+import org.springframework.batch.core.configuration.annotation.StepBuilderFactory;
+import org.springframework.batch.core.configuration.annotation.StepScope;
+import org.springframework.batch.integration.async.AsyncItemProcessor;
+import org.springframework.batch.integration.async.AsyncItemWriter;
+import org.springframework.batch.item.ItemProcessor;
+import org.springframework.batch.item.ItemStreamReader;
+import org.springframework.batch.item.ItemStreamWriter;
+import org.springframework.batch.item.ItemWriter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+
+
+/**
+ *
+ * @author ochoaa
+ */
+@Configuration
+@EnableBatchProcessing
+@EnableAsync
+@ComponentScan(basePackages = "org.mskcc.cmo.messaging")
+public class BatchConfiguration {
+
+    public static final String LIMS_REQUEST_PUBLISHER_JOB = "limsRequestPublisherJob";
+
+    @Autowired
+    public JobBuilderFactory jobBuilderFactory;
+
+    @Autowired
+    public StepBuilderFactory stepBuilderFactory;
+
+    @Autowired
```

Comment:
Do we need this here if its in the writer?

Reply:
it needs to be initialized first which is why it's here as well (see L52)

Reply:
Which line 52 are you referring to?  I only see the reference used here:
[LimsRequestWriter](https://github.com/mskcc/cmo-metadb-sample-publisher/blob/926023c5e56837f4c394ed2981dbb610c6720811/src/main/java/org/mskcc/cmo/publisher/pipeline/limsrest/LimsRequestWriter.java#L21).

Reply:
@n1zea144 sorry, meant to link it [here ](https://github.com/mskcc/cmo-metadb-sample-publisher/pull/14/files#diff-5028471369ee9aa5329478ec6110327743aeb2ea1e5d5d326b2adb4c1a8e6e80R68)

Reply:
Can you not autowire initGatewayConnection in the writer?